extensions/amp-story/1.0/amp-story.js

```diff
@@ -1068,10 +1064,12 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   getInitialPageId_(firstPageEl) {
-    const historyPage = /** @type {string} */ (getHistoryState(
-      this.win,
-      HistoryState.PAGE_ID
-    ));
+    const pages =
+      /**  @type {!Array} */ (getHistoryState(
+        this.win,
+        HistoryState.NAVIGATION_PATH
+      ) || []);
+    const historyPage = lastItem(pages);
```

Comment:
Nit: not your doing, but should this block be after the the following block that looks for the `page` query parameter?